flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartition.java

```diff
@@ -93,12 +92,8 @@
 	/**
 	 * Common constructor.
 	 */
-	public BoundedBlockingSubpartition(
-			int index,
-			ResultPartition parent,
-			Path filePath) throws IOException {
-
-		this(index, parent, MemoryMappedBuffers.create(filePath));
+	BoundedBlockingSubpartition(int index, ResultPartition parent) throws IOException {
+		this(index, parent, MemoryMappedBuffers.create(parent.createFileChannelPath()));
```

Comment:
`MemoryMappedBuffers.create(parent.createFileChannelPath())`:
I think all complicated construction logic should happen in `ResultPartitionFactory`. `ResultPartition` should not provide convenience methods for construction.

Reply:
I could refactor `parent.createFileChannelPath()` into factory while creating subpartitions and pass the path in the constructor directly